src/basic/path-util.c

```diff
@@ -690,6 +709,22 @@ int find_executable_full(const char *name, bool use_path_envvar, char **ret_file
                 if (!path_extend(&element, name))
                         return -ENOMEM;
 
+                if (root) {
+                        char *path_name = NULL;
```

Comment:
style nit: missing new line after declaration, and initializing to NULL is no longer necessary